python/lsst/pipe/drivers/multiBandDriver.py

```diff
@@ -160,7 +162,29 @@ def __init__(self, butler=None, schema=None, refObjLoader=None, reuse=tuple(), *
         self.reuse = tuple(reuse)
         self.makeSubtask("detectCoaddSources")
         self.makeSubtask("mergeCoaddDetections", schema=schema)
-        self.makeSubtask("measureCoaddSources", schema=afwTable.Schema(self.mergeCoaddDetections.schema),
+        if self.config.measureCoaddSources.inputCatalog.startswith("deblended"):
+            # Ensure that the output from deblendCoaddSources matches the input to measureCoaddSources
+            self.measurementInput = self.config.measureCoaddSources.inputCatalog
+            self.deblenderOutput = []
+            if self.config.deblendCoaddSources.simultaneous:
+                if self.config.deblendCoaddSources.multiBandDeblend.conserveFlux:
+                    self.deblenderOutput.append("deblendedFlux")
+                if self.config.deblendCoaddSources.multiBandDeblend.saveTemplates:
+                    self.deblenderOutput.append("deblendedModel")
+            else:
+                self.deblenderOutput.append("deblendedFlux")
+            if self.measurementInput not in self.deblenderOutput:
+                err = "Measurement input '{0}' is not in the list of deblender output catalogs '{1}'"
+                raise ValueError(err.format(self.measurementInput, self.deblenderOutput))
+
+            self.makeSubtask("deblendCoaddSources",
+                             schema=afwTable.Schema(self.mergeCoaddDetections.schema),
+                             peakSchema=afwTable.Schema(self.mergeCoaddDetections.merged.getPeakSchema()),
+                             butler=butler)
+            measureInputSchema = afwTable.Schema(self.deblendCoaddSources.schema)
+        else:
+            measureInputSchema = afwTable.Schema(self.mergeCoaddDetections.schema)
```

Comment:
Does this if/else exist to make this new multiBandDriver backward-compatible with older data products generated with an older Stack? 

In other words, for running `multiBandDriver.py` consistently with a post-DM-15104 Stack, shouldn't `self.config.measureCoaddSources.inputCatalog.startswith("deblended")` always be true?

Reply:
On second look, maybe this part was added to make `multiBandDriver` more configurable?

Reply:
No, this is not always true. Just like before when it was possible to turn the deblender off with `doDeblend`, it is now possible to skip deblending using `inputCatalog="mergeDet"`.